packages/fxa-profile-server/test/api.js

```diff
@@ -1685,7 +1685,27 @@ describe('api', function () {
         return hash.update(anonId).digest('hex');
       }
 
-      xit('should post a new ecosystem_anon_id', function () {});
+      it('should put a new ecosystem_anon_id', async function () {
+        mock.anonIdUpdated();
+
+        mock.token({
+          user: USERID,
+          scope: ['profile:ecosystem_anon_id:write'],
+        });
+
+        const res = await Server.api.post({
+          url: '/ecosystem_anon_id',
+          payload: {
+            ecosystemAnonId: '12345',
+          },
+          headers: {
+            authorization: 'Bearer ' + tok,
+          },
+        });
+
+        assert.equal(res.statusCode, 200);
```

Comment:
This is good and all, and means we made it to the end of the request without any errors, but I'm not sure how to assert request body or headers to the auth server, with how we're [stubbing requests](https://github.com/mozilla/fxa/pull/6062/files#diff-7af5e8103c0a67bff380affec012e4c1R149). Any ideas?

Reply:
> how to assert request body or headers to the auth server

Would you be able to spy on the nock object?

Reply:
It looks like nock can return a specific response only if headers match ([`.matchHeader()`](https://www.npmjs.com/package/nock#request-headers-matching)) or only if the [body matches](https://www.npmjs.com/package/nock#specifying-request-body). @jodyheavener is that what you were looking for?

Reply:
Ahh, I get it now. I was looking for a way to reactively inspect the request, but it looks like in Nock's case the route is simply not called if _all_ the setup requirements are not met. So a successful call == the expected values have been met. Thanks!